From a223da7de5a9c1d842ecadad38556174d130d803 Mon Sep 17 00:00:00 2001
From: Meng Wei <wei.meng@intel.com>
Date: Mon, 3 Dec 2018 16:53:04 +0800
Subject: [PATCH 0988/1214] media: buttress: Use ISR_STATUS in buttress ISR

Use ISR_STATUS register instead of ISR_ENABLE_STATUS register to
clear all interrupts in buttress ISR.

Change-Id: Icc587a160355b5f6f1c884b673f23e01b4ef5b2d
Tracked-On: PKT-1732
Signed-off-by: Hongyu Yi <hongyu.yi@intel.com>
Signed-off-by: Meng Wei <wei.meng@intel.com>
---
 drivers/media/pci/intel/ipu-buttress.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/media/pci/intel/ipu-buttress.c b/drivers/media/pci/intel/ipu-buttress.c
index b823c0f..d19e965 100644
--- a/drivers/media/pci/intel/ipu-buttress.c
+++ b/drivers/media/pci/intel/ipu-buttress.c
@@ -404,10 +404,10 @@ irqreturn_t ipu_buttress_isr(int irq, void *isp_ptr)
 	irqreturn_t ret = IRQ_NONE;
 	u32 disable_irqs = 0;
 	u32 irq_status;
-#ifdef CONFIG_VIDEO_INTEL_IPU4P
-	u32 reg_irq_sts = BUTTRESS_REG_ISR_STATUS;
-#else
+#ifdef CONFIG_VIDEO_INTEL_IPU4
 	u32 reg_irq_sts = BUTTRESS_REG_ISR_ENABLED_STATUS;
+#else
+	u32 reg_irq_sts = BUTTRESS_REG_ISR_STATUS;
 #endif
 	unsigned int i;
 
-- 
2.7.4

